tests/mdev_configuration_allocation_test.go

```diff
@@ -116,6 +118,10 @@ var _ = Describe("[Serial][sig-compute]MediatedDevices", func() {
 
 		AfterEach(func() {
 			cleanupConfiguredMdevs()
+			if singleNode.Name != "" {
```

Comment:
I think this approach is error-prone. You're assuming that every test will set this variable if it adds `mdevTestLabel` label. Someone could easily add another test and do something like:

```go
node := util.GetAllSchedulableNodes(virtClient).Items[0]
```

This will add the label but cause it to not be cleaned. I think that the best approach would be to save each node's labels then recover them after tests are finished. This way we don't have to maintain this cleanup in the future (declare variables for the labels we want to delete, declare which node was used during the test, etc').

Something like:
```go
var nodesLabels // Some structure, map probably

...

BeforeEach(func() {
    nodesLabels = backupNodes()
})

...

AfterEach(func() {
    backupNodes(nodesLabels)
})
```

Reply:
BTW, If we want to take it a step further, this probably better be done [here](https://github.com/kubevirt/kubevirt/blob/main/tests/utils.go#L500). This way we could recover all labels for every test without any further help.

Reply:
Thanks @iholder-redhat for the clean review. You are definitely right, I'm going to change it

Reply:
Thanks @fossedihelm!

Another thought that came into my mind: it might be problematic running multiple tests in parallel that change / depend on node's annotations.

If we go with the "BeforeAll/AfterAll" approach I think we also need to mark the tests to run in serial (i.e. `[Serial]`). If you do it [here](https://github.com/kubevirt/kubevirt/blob/main/tests/utils.go#L500) it might be a bit safer in the aspect of after-test-cleaning.

Reply:
My approach would have been based on standardizing the label applied to the node. Basically, the leftover was caused by the name of the label. Since the `SynchronizedAfterTestSuiteCleanup` function already delete the labels which has `tests.kubevirt.io` prefix, I would change the name of the applied label (something like [this](https://github.com/kubevirt/kubevirt/blob/bb05152cd6abaf48e60e2474287f1f58f22465ed/tests/migration_test.go#L3051)) What do you think?

Reply:
In any case, this test is already marked as `[Serial]` :slightly_smiling_face:

Reply:
> My approach would have been based on standardizing the label applied to the node. Basically, the leftover was caused by the name of the label. Since the SynchronizedAfterTestSuiteCleanup function already delete the labels which has tests.kubevirt.io prefix, I would change the name of the applied label (something like this) What do you think?

Sounds good to me, but wouldn't it be better to simply recover all the labels that were defined before test suite started to run?

> In any case, this test is already marked as [Serial] :slightly_smiling_face:

Aweomse :)